lua/cfc_time/server/time_tracking.lua

```diff
@@ -132,28 +135,41 @@ function ctime:initPlayer( ply )
             initialTime.seconds = seconds
         end
 
-        hook.Run( "CFC_Time_PlayerInitialTime", ply, isFirstVisit, initialTime )
+        hook.Run( "CFC_Time_PlayerInitialTime", ply, isFirstVisit, initialTime, lastSession )
         sessionTotalTime = sessionTotalTime + initialTime.seconds
 
         ctime.totalTimes[steamID] = sessionTotalTime
         ctime:broadcastPlayerTime( ply, sessionTotalTime, now, 0 )
     end
 
-    storage:PlayerInit( ply, now, function( data )
-        if not IsValid( ply ) then return end
+    local existingSession = self.sessions[steamID]
+    if existingSession then
+        ErrorNoHaltWithStack( "[CFCTime] Player loaded in, but already had a session? " .. steamID )
+        self:untrackPlayer( steamID )
+    end
```

Comment:
New safeguard. While exceedingly unlikely, it was technically possible for a player to load in before their previous session was marked as departed and saved to the DB.

This doesn't fix that problem (I don't know how) but we'll at least know about it now.